Libraries/dotNetRDF/Writing/TurtleWriter.cs

```diff
@@ -182,21 +167,60 @@ private void GenerateOutput(TurtleWriterContext context)
 
                 // Get the Triples as a Sorted List
                 List<Triple> ts = context.Graph.Triples.ToList();
-                ts.Sort(new FullTripleComparer(new FastNodeComparer()));
+                int capacity = ts.Count;
+                Dictionary<INode, Dictionary<INode, Dictionary<INode, List<Triple>>>> sortHelperDictionary = new Dictionary<INode, Dictionary<INode, Dictionary<INode, List<Triple>>>>(capacity);
+                // Fill dictionary
+                foreach (Triple triple in ts)
+                {
+                    if (!sortHelperDictionary.ContainsKey(triple.Subject)) { sortHelperDictionary.Add(triple.Subject, new Dictionary<INode, Dictionary<INode, List<Triple>>>()); }
+
+                    if (!sortHelperDictionary[triple.Subject].ContainsKey(triple.Predicate)) { sortHelperDictionary[triple.Subject].Add(triple.Predicate, new Dictionary<INode, List<Triple>>()); }
+
+                    if (!sortHelperDictionary[triple.Subject][triple.Predicate].ContainsKey(triple.Object)) { sortHelperDictionary[triple.Subject][triple.Predicate].Add(triple.Object, new List<Triple>()); }
```

Comment:
I think this level of the dictionary is not really needed. Objects don't have to be output in sort order, we just need the ordering to allow grouping by subject and predicate.